articlequality/extractors/ptwiki.py

```diff
@@ -6,24 +6,48 @@
 logger = logging.getLogger(__name__)
 
 
+POSSIBLE_LABELS = {
+    "1": "1",
+    "2": "2",
+    "3": "3",
+    "4": "4",
+    "5": "5",
+    "6": "6",
+    "ab": "5",
+    "ad": "6"
+}
+
+
 def from_template(template):
-    project_name = normalize_project_name(template.name)
-    if project_name == "marca de projeto":
-        labels = extract_labels(template)
-        if len(labels) >= 1:
-            for project, label in labels[1:]:
-                yield (normalize_project_name(project), label)
+    template_name = normalize_template_name(template.name)
+    if template_name == "marca de projeto":
+        label = extract_label(template)
+        if label is not None:
+            yield ("marca de projeto", label)
+        else:
+            logger.warn("Could not extract label from {0}".format(str(template)))
 
 
-PROJECT_LABEL = re.compile(r"([^\|\{\{\}\}]+)\|([0-5\*])", re.I)
+def extract_label(template):
+    try:
+        label = template.get("qualidade").value
```

Comment:
For consistency with [w:pt:Module:Avaliação](https://pt.wikipedia.org/wiki/M%C3%B3dulo:Avalia%C3%A7%C3%A3o), I would try first to get the parameter "1" and then, if that fails, try its alias "qualidade". Although, I don't think there is any case where users added both names with different values, so it should not matter in practice.

Reply:
Ha!  I tried that but it fails on this example:
`{{Marca de projeto|qualidade=5|Foobar|5}}`

When we look at parameter "1", we get "Foobar".

Reply:
Yep, which should be treated as `None`, since `foobar` is not one of the possible labels (valid quality levels). While
`{{Marca de projeto|qualidade=5|3|foobar}}`
and
`{{Marca de projeto|3|foobar|qualidade=5}}`
should return the label 3.

Reply:
`{{Marca de projeto|qualidade=5|3|foobar}}` is super weird, but OK!